config/docker/dependencies/ubuntu22/openmpi/Dockerfile

```diff
@@ -11,8 +11,8 @@ RUN wget https://apt.kitware.com/kitware-archive.sh &&\
     sh kitware-archive.sh
 
 RUN export DEBIAN_FRONTEND=noninteractive &&\
-    apt-get install gcc g++ \ 
-    clang \
+    apt-get install gcc-9 g++-9 \ 
+    clang-14 \
```

Comment:
After this, do gcc and g++ automatically point to 9 and does clang point to 14?

Reply:
Since we are doing versioned install, please do clang-format-14 clang-tidy-14 and libomp-14-dev

Reply:
[these lines ](https://github.com/QMCPACK/qmcpack/pull/4696/files#diff-3d90db0e077b1e8d01e74c131d44a4b816cef04c293c7ef57e14d3aea5d18d8fR52-R58)swap the defaults to the new versions